Legend on/off

elpaso opened the pull request:
LEGEND_ON=<layer_id>:<rule_key>,<rule_key>;<layer_id>:<rule_key>,<rule_key> 
LEGEND_OFF=<layer_id>:<rule_key>,<rule_key>;<layer_id>:<rule_key>,<rule_key>

Fixes #407

elpaso committed:
Legend on/off

LEGEND_ON=<layer_id>:<rule_key>,<rule_key>;<layer_id>:<rule_key>,<rule_key>
LEGEND_OFF=<layer_id>:<rule_key>,<rule_key>;<layer_id>:<rule_key>,<rule_key>

volterra79 commented:
@elpaso for g3w-client purpose, it will be mandatory attribute **categories** in layer configuration from project config api.
If a layer has categorties it will be filled otherwise will be undefined.
Example: 
{
    "id": "buildings_2f43dc1d_6725_42d2_a09b_dd446220104a",
    "name": "buildings",
    "origname": "buildings",
    "geometrytype": "MultiPolygon",
    "crs": {
        "epsg": 3857,
        "proj4": "+proj=merc +a=6378137 +b=6378137 +lat_ts=0 +lon_0=0 +x_0=0 +y_0=0 +k=1 +units=m +nadgrids=@null +wktext +no_defs",
        "geographic": false,
        "axisinverted": false
    },
    "title": "buildings",
    "scalebasedvisibility": false,
   ...
   **categories: [{....}]**
}

elpaso commented:
> @elpaso for g3w-client purpose, it will be mandatory attribute **categories** in layer configuration from project config api. If a layer has categorties it will be filled otherwise will be undefined. Example: { "id": "buildings_2f43dc1d_6725_42d2_a09b_dd446220104a", "name": "buildings", "origname": "buildings", "geometrytype": "MultiPolygon", "crs": { "epsg": 3857, "proj4": "+proj=merc +a=6378137 +b=6378137 +lat_ts=0 +lon_0=0 +x_0=0 +y_0=0 +k=1 +units=m +nadgrids=@null +wktext +no_defs", "geographic": false, "axisinverted": false }, "title": "buildings", "scalebasedvisibility": false, ... **categories: [{....}]** }

ok, hold on this PR until I add this piece of implementation.

elpaso committed:
Add class info to GetLegendGraphic

volterra79 committed:
Client:
Test client

volterra79 commented:
@elpaso update static client files. But it seem that doesn't work properly. If uncheck all categories of a layer the layer disappear, if i uncheck only one category, nothing happen and the request seems ok.

Commit:
Typo

Commit:
Fix redirect url on update map group from group's project list.

elpaso committed:
Handle multithreading issue and layer_id/name

Commit:
Remove django-debug-panel from requirements.

volterra79 committed:
Client:
https://github.com/g3w-suite/g3w-client/tree/legend_categories

volterra79 committed:
Client:
https://github.com/g3w-suite/g3w-client/tree/legend_categories

volterra79 committed:
Client:
https://github.com/g3w-suite/g3w-client/commit/94aa1169b664ec2469a5b7baac653913cb0cc0a7

volterra79 committed:
Client:
https://github.com/g3w-suite/g3w-client/pull/108/commits/30539df5ce01ccdde59416ade03f213b2dae322e

volterra79 committed:
Client:
https://github.com/g3w-suite/g3w-client/pull/108/commits/7a7d8c3970711735430ebc57d8298fa6a0eec98c

Commit:
Merge branch 'dev' into legend-classes-onoff

# Conflicts:
#	g3w-admin/client/static/client/css/app.min.css
#	g3w-admin/client/static/client/js/app.min.js
